tests/unit/lms/db/_bulk_action_test.py

```diff
@@ -145,3 +93,22 @@ class MisconfiguredModel:  # pylint: disable=unused-variable
                 NOT_THE_RIGHT_NAME = BulkAction.Config(
                     upsert_index_elements=["id"], upsert_update_elements=["name"]
                 )
+
+    def assert_has_rows(self, db_session, *attrs):
+        rows = list(db_session.query(self.TableWithBulkUpsert))
+
+        assert (
+            rows
+            == Any.iterable.containing(
+                [
+                    Any.instance_of(self.TableWithBulkUpsert).with_attrs(expected)
+                    for expected in attrs
+                ]
+            ).only()
+        )
+
+    @pytest.fixture
+    def with_upsert_trigger_onupdate(self):
+        self.TableWithBulkUpsert.BULK_CONFIG.upsert_trigger_onupdate = True
+        yield
+        self.TableWithBulkUpsert.BULK_CONFIG.upsert_trigger_onupdate = False
```

Comment:
If we don't like this, it might be possible to get what we want by sub-classing the other one? Not sure. Either way, we might want to pick shorter names for the tables, as that really bloats the line length by causing black to insert a lot of new lines in certain cases.

We can probably get what type of table this is from context. So it could probably be `TestModel` without losing too much.